### PR TITLE
Update plugin export docs

### PR DESCRIPTION
**Pull request by timja**
### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!

- [x] Please describe what you did

- [x] Link to relevant GitHub issues or pull requests

- [x] Link to relevant [Jenkins JIRA issues](https://issues.jenkins-ci.org)

- [na] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### Description

Added new util methods for getting root elements and updated plugin developer guide for unit testing exports

### CODE REVIEW

File: docs/PLUGINS.md [hunk 1]
```diff
@@ -277,24 +277,16 @@ You also can write a test case to check export from a live instance is well supp
...
-        
-        List<CNode> yourList = mapping.get("a-list-property").asSequence();
-        assertEquals("list size", 5, yourList.size());
-        assertEquals(yourList.get(0).getScalar().getValue(), "expected-value");
-        assertEquals(yourList.get(0).asMapping().getScalarValue("key"), "expected-key");
+        final CNode yourAttribute = getUnclassifiedRoot(context).get("<your-attribute>");
+
+         String exported = toYamlString(yourAttribute);
+
+         String expected = new String(Files.readAllBytes(Paths.get(getClass()
```
Comment (casz) [inline docs/PLUGINS.md]: Perhaps provide another util method?
Assumes it should use `getClass().getResource()` unless String path is provided?

Comment (casz) [reply docs/PLUGINS.md]: Actually, the configured as code annotation could use the very same method :laughing:

Comment (timja) [reply docs/PLUGINS.md]: not sure what you mean?
There's Resources.toString("") in guava which could be used instead but wasn't 100% sure if guava was everywhere

Comment (casz) [reply docs/PLUGINS.md]: Just no to guava :) I'll submit my suggestion during the weekend.

Comment (casz) [reply docs/PLUGINS.md]: What I meant I am Saying create a method for `toStringFromYamlFile` to shorten the whole `Files.readAllBytes(Paths...(getClass()...)` nonsense

Comment (timja) [reply docs/PLUGINS.md]: I can't find a way that works in a static method in another project without using guava or spring

The classloaders aren't correct...

Comment (casz) [reply docs/PLUGINS.md]: You have access to Junit methods that should take care of the same.

Comment (timja) [reply docs/PLUGINS.md]: I can't see any special junit methods for it?
The ConfiguredWithCodeRule has access to the class in the users test so can use the correct class loader

Comment (casz) [reply docs/PLUGINS.md]: `Thread.currentThread().getStackTrace()[2].getClass`?
Rest should work

Comment (timja) [reply docs/PLUGINS.md]: that doesn't work in tests in jcasc...

Comment (casz) [reply docs/PLUGINS.md]: `toStringFromYamlFile(this, "my.yml")`?

Comment (timja) [reply docs/PLUGINS.md]: done that 👍

File: plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java [hunk 2]
```diff
@@ -35,4 +60,14 @@ public static String toYamlString(CNode rootNode) throws IOException {
         serializeYamlNode(yamlRoot, buffer);
         return buffer.toString();
     }
+
+    public static String toStringFromYamlFile(Object clazz, String resourcePath) throws URISyntaxException, IOException {
```
Comment (casz) [inline plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java]: We should add some java docs to all of these? To all of misc?

Comment (timja) [reply plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java]: done

File: plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java [hunk 3]
```diff
@@ -1,38 +1,142 @@
...
+     * @return the jenkins root mapping
+     * @throws Exception something's not right...
+     */
     public static Mapping getJenkinsRoot(ConfigurationContext context)
         throws Exception {
         JenkinsConfigurator root = getJenkinsConfigurator();
         return Objects.requireNonNull(root.describe(root.getTargetComponent(context), context)).asMapping();
     }
 
-    public static Mapping getJenkinsRoot(JenkinsConfigurator root, ConfigurationContext context)
```
Comment (timja) [inline plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java]: I couldn't see any reason for anyone to use this? and there doesn't seem to be any usages of it?

Can restore it if you see a need

Comment (casz) [reply plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java]: Fine with me

**Pull request merged**